include/zephyr/net/ethernet.h

```diff
@@ -347,6 +347,19 @@ enum ethernet_filter_type {
 
 /** @endcond */
 
+/** Types of Ethernet L2 */
+enum ethernet_if_types {
+	/** IEEE 802.3 Ethernet */
+	L2_ETH_IF_TYPE_ETHERNET			= BIT(0),
```

Comment:
Although not likely to ever be used - wouldn't defining type like this (i. .e bitfields) require changes in **all** existing Ethernet interfaces? Otherwise they end up with unspecified `eth_if_type`. I still think though, that Ethernet should be implicit and Wi-Fi should just get a flag (or capability field).

Reply:
I have now removed bit fields, so, left the enum with a comment that this is default.